packages/core/src/routes/hook.ts

```diff
@@ -23,11 +26,25 @@ export default function hookRoutes<T extends AuthedRouter>(
 
   router.post(
     '/hooks',
-    koaGuard({ body: Hooks.createGuard.omit({ id: true }), response: Hooks.guard, status: 200 }),
+    koaGuard({
+      body: Hooks.createGuard
+        .omit({ id: true, event: true, enabled: true })
+        .extend({ config: hookConfigGuard.omit({ signingKey: true }) }),
+      response: Hooks.guard,
+      status: [200, 422],
+    }),
     async (ctx, next) => {
+      const { config, events, ...rest } = ctx.guard.body;
+      assertThat(events.length > 0, new RequestError({ code: 'hook.empty_events', status: 422 }));
+
       ctx.body = await insertHook({
+        ...rest,
         id: generateStandardId(),
-        ...ctx.guard.body,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        event: events[0]!, // Note: the `event` is deprecated, but we still need to set it for backward compatibility.
```

Comment:
throw an error (without i18n) would be better then `!`, actually it is more convenient then disable lint.

Reply:
Updated.